src/DotNetWorker.Grpc/GrpcFunctionsHostLoggerProvider.cs

```diff
@@ -5,20 +5,33 @@
 using Microsoft.Azure.Functions.Worker.Logging;
 using Microsoft.Azure.Functions.Worker.Grpc.Messages;
 using Microsoft.Extensions.Logging;
+using Azure.Core.Serialization;
+using Microsoft.Extensions.Options;
 
 namespace Microsoft.Azure.Functions.Worker.Diagnostics
 {
     internal class GrpcFunctionsHostLoggerProvider : ILoggerProvider, ISupportExternalScope
     {
         private readonly ChannelWriter<StreamingMessage> _channelWriter;
+        private readonly ObjectSerializer _serializer;
         private IExternalScopeProvider? _scopeProvider;
 
-        public GrpcFunctionsHostLoggerProvider(GrpcHostChannel outputChannel)
+        public GrpcFunctionsHostLoggerProvider(GrpcHostChannel outputChannel, IOptions<WorkerOptions> workerOptions)
         {
             _channelWriter = outputChannel.Channel.Writer;
+
+            if (workerOptions?.Value?.Serializer != null)
+            {
+                _serializer = workerOptions.Value.Serializer;
+            }
+            else
+            {
+                _serializer = new JsonObjectSerializer();
```

Comment:
[In other places, we throw](https://github.com/Azure/azure-functions-dotnet-worker/blob/main/src/DotNetWorker.Core/Converters/JsonPocoConverter.cs#L26) when workerOptions.Serializer is `null`.